### ISSUE TITLE
about trainset.jsonl

### ISSUE THREAD

**Issue by Devil-Ideal** [zhaoziheng/SAT#11]
Thank you for your outstanding work. Could you please provide the trainset.jsonl file and the related code?

**Comment by zhaoziheng** [comment]
You mean how we split those datasets and process them ? You can find the jsonl for each dataset and the preprocessing code in this repo: https://github.com/zhaoziheng/SAT-DS/tree/main. To get the 'trainset.jsonl' of all the 72 datasets , just merge them into one.

**Comment by Devil-Ideal** [comment]
> You mean how we split those datasets and process them ? You can find the jsonl for each dataset and the preprocessing code in this repo: https://github.com/zhaoziheng/SAT-DS/tree/main. To get the 'trainset.jsonl' of all the 72 datasets , just merge them into one.

Here’s the translation:

"I noticed in `trainer.py`:

`text, modality, image, mask, dataset_name, query_mask = batch['text'], batch['modality'], batch['image'], batch['mask'], batch['dataset'], batch['query_mask']`

However, in `train_dataset.py`, `getitem` only returns:

`return {'image': image, 'mask': mask, 'text': chosen_label, 'modality': modality, 'image_path': sample['renorm_image'], 'mask_path': sample['renorm_segmentation_dir'], 'dataset': sample['dataset'], 'y1x1z1_y2x2z2': y1x1z1_y2x2z2}`

What is the reason for this inconsistency?"

**Comment by zhaoziheng** [comment]
The `getitem` in `train_dataset.py` will return a sample. However, I have to do something like padding to align all the samples in a batch, this is realized in `collect_fn.py`. In this case, I create `query_mask` to indicate the padded text prompts.
